### ISSUE TITLE
vcfallelicprimitives misrepresenting deletions

### ISSUE THREAD

**Issue by jpuritz** [vcflib/vcflib#83]
Hey Erik,

I found an error in vcfallelicprimitives.  It's basically misrepresenting a deletion when it's compounded with a MNP (or possibly other variants).

This is the original vcf
E757_L102   203 .   CAC CC,CCG  102888  .   AB=0.510373,0.482877;ABP=3.46081,3.75395;AC=48,21;AF=0.6,0.2625;AN=80;AO=623,335;CIGAR=1M1D1M,1M2X;DP=1159;DPB=951.333;DPRA=0.310632,0.898558;EPP=1355.84,730.454;EPPR=437.305;GTI=0;LEN=1,2;MEANALT=1.44118,1.73684;MQM=60,60;MQMR=60;NS=40;NUMALT=2;ODDS=0;PAIRED=1,1;PAIREDR=1;PAO=0,0;PQA=0,0;PQR=0;PRO=0;QA=20559,11055;QR=6600;RO=200;RPL=623,335;RPP=1355.84,730.454;RPPR=437.305;RPR=0,0;RUN=1,1;SAF=0,0;SAP=1355.84,730.454;SAR=623,335;SRF=0;SRP=437.305;SRR=200;TYPE=del,mnp;technology.Illumina=1,1 GT:DP:RO:QR:AO:QA:GL    0/1:28:14:462:14,0:462,0:-145.47,0,-145.47,-149.685,-149.685,-299.039

And these are the primative bases:
E757_L102   203 .   CA  C   102888  .   AC=48;AF=0.6;LEN=1;TYPE=del GT  0|1
[jpuritz@gdcsrv1 sim]$ mawk '/E757_L102\t204/' T.vcf | cut -f1-9,49
E757_L102   204 .   A   C   102888  .   AC=21;AF=0.2625;LEN=1;TYPE=snp  GT  0|0
[jpuritz@gdcsrv1 sim]$ mawk '/E757_L102\t205/' T.vcf | cut -f1-9,49
E757_L102   205 .   C   G   102888  .   AC=21;AF=0.2625;LEN=1;TYPE=snp  GT  0|0

Basically, the GT for 204 should be null or missing.

Thanks!

Jon

**Comment by tseemann** [comment]
Complex alleles like this + flat VCF files = make me want to go hide in a cave!
